k256/Cargo.toml

```diff
@@ -24,6 +24,7 @@ sec1 = { version = "0.2", default-features = false }
 
 # optional dependencies
 ecdsa-core = { version = "0.13", package = "ecdsa", optional = true, default-features = false, features = ["der"] }
+hash2field = { version = "0.4", optional = true }
```

Comment:
@tarcieri Maybe it's worth to incorporate this crate into `elliptic-curve`?

Reply:
@mikelodder7 would you be interested in upstreaming `hash2field` somewhere, either `elliptic-curve` or possibly `ff`?

Reply:
Sure thing. Elliptic-curve probably makes more sense but I can be convinced otherwise

Reply:
Sounds good!

If you're up for it, I think it might also be possible to extract a generic implementation of SSWU, although I'm fine with duplication for now and I can try to extract it as part of a follow-up:

https://github.com/RustCrypto/elliptic-curves/issues/439#issuecomment-929500577

You can use e.g. `ff::{Field, PrimeField}` as the bounds, and I can look into ensuring they're impl'd for the respective `FieldElement` types in `k256` and `p256`.

Reply:
Yeah if `ff::Field` is impl'd then the dedup is easy from there. `osswu` will just have to take a set of params but otherwise, they'll be the same. Should `osswu` go upstream as well in `elliptic-curve`?

Reply:
I think it'd be great if you can add a generic implementation, and in parallel I can work on trying to get the `ff` traits impl'd for the `FieldElement` types for `k256` and `p256`.

Reply:
Cool. I'd extract the `osswu` method and `hash2field` crate to `elliptic-curve`.

Reply:
Awesome!